src/sortStudents.ts

```diff
@@ -1,16 +1,95 @@
+// import { type } from "os";
 
 export interface Student {
-  // describe Student interface
+  name: string,
+  surname: string,
+  age: number,
+  married: boolean,
+  grades: number[],
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'grades',
 }
 
 // create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+export function sortStudents(students: Student[], sortBy: SortType,
+  order: SortOrder): Student[] {
+  const copyOfStudents: Student[] = [];
 
-export function sortStudents(students, sortBy, order) {
-  // write your function
+  students.forEach((item) => copyOfStudents.push({ ...item }));
+
+  switch (sortBy) {
+    case SortType.Name:
+    case SortType.Surname:
+      copyOfStudents.sort((a: Student, b: Student): number => {
+        if (order === 'asc') {
+          return a[sortBy] >= b[sortBy]
+            ? 1
+            : -1;
+        }
+
+        if (order === 'desc') {
+          return b[sortBy] >= a[sortBy]
+            ? 1
+            : -1;
+        }
+
+        return 0;
+      });
+      break;
+
+    case SortType.Married:
+    case SortType.Age:
+      copyOfStudents.sort((a: Student, b: Student): number => {
+        if (order === 'asc') {
+          return a[sortBy] >= b[sortBy]
+            ? 1
+            : -1;
+        }
+
+        if (order === 'desc') {
+          return b[sortBy] >= a[sortBy]
+            ? 1
+            : -1;
+        }
+
+        return 0;
```

Comment:
here you can use simple mathematics instead. (just use something to make number out of boolean)
In suggestion plusses are doing type conversion.
```suggestion
if (order === 'asc') {
    return +a[sortBy] - +b[sortBy];
  }
  
return +b[sortBy] - +a[sortBy];
```

Reply:
good trick to avoid ts errors ))